server/src/main/java/io/crate/metadata/view/ViewInfo.java

```diff
@@ -44,22 +44,25 @@ public class ViewInfo implements RelationInfo {
 
     private final RelationName ident;
     private final String definition;
-    private final List<Reference> columns;
+    private final List<Reference> references;
     private final String owner;
     private final SearchPath searchPath;
 
     @VisibleForTesting
-    public ViewInfo(RelationName ident, String definition, List<Reference> columns, @Nullable String owner, SearchPath searchPath) {
+    public ViewInfo(RelationName ident, String definition, List<Reference> references, @Nullable String owner, SearchPath searchPath) {
         this.ident = ident;
         this.definition = definition;
-        this.columns = columns;
+        this.references = references;
```

Comment:
We could create the needed `columns` (only top columns, sorted by position) and `references` (all references, sorted by position) already at the ctor to avoid processing when their getters are called

Reply:
Pushed an update, thx